### PR TITLE
Modernize and release

### PR DESCRIPTION
**Pull request by alejoe91**
@lepmik @nicolossus same as spatial map :)

@lepmik can you add the `PYPI_API_TOKEN` here as well?

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,59 @@
+[project]
+name = "head_direction"
+version = "0.1.0"
+authors = [
+    { name = "Mikkel Lepperod", email = "mikkel@simula.no" },
+    { name = "Alessio Buccino", email = "alessiop.buccino@gmail.com" },
+]
+
+description = "Compute spatial maps for neural data."
```
Comment (nicolossus) [inline pyproject.toml]:
```suggestion
description = "Analysis of head direction cells."
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      - id: requirements-txt-fixer
+      - id: detect-private-key
+      - id: check-merge-conflict
+
+  - repo: https://github.com/psf/black
+    rev: 24.4.2
+    hooks:
+      - id: black
+        exclude: tests/test_data
+      - id: black-jupyter
```
Comment (nicolossus) [inline .pre-commit-config.yaml]:
```suggestion
      - id: end-of-file-fixer
      - id: check-docstring-first
      - id: debug-statements
      - id: check-toml
      - id: check-yaml
      - id: requirements-txt-fixer
      - id: detect-private-key
      - id: check-merge-conflict

  - repo: https://github.com/psf/black
    rev: 24.4.2
    hooks:
      - id: black
      - id: black-jupyter
```

The excludes are not needed here.

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+readme = "README.md"
+requires-python = ">=3.10"
+classifiers = [
+    "Programming Language :: Python :: 3",
+    "License :: OSI Approved :: MIT License",
+    "Operating System :: OS Independent",
+]
+
+dependencies = [
+    "numpy<2",
```
Comment (nicolossus) [inline pyproject.toml:19]: Won't this quickly create a conflict since many packages now are based on numpy>=2?

**Review by nicolossus — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 3 above)
Comment (alejoe91) [reply pyproject.toml:19]: I think several packages are still lagging behind due to updates on the numba side (e.g. SpikeInterface)

**Review by nicolossus — APPROVED** [review]

**Comment by lepmik** [comment]
api token done

**Pull request merged**